modules/tinymce/src/plugins/advlist/main/ts/core/ListUtils.ts

```diff
@@ -1,18 +1,24 @@
-import { Optional, Type } from '@ephox/katamari';
+import { Arr, Optional, Type } from '@ephox/katamari';
 
 import Editor from 'tinymce/core/api/Editor';
 
+const isCustomList = (list: Element): boolean =>
+  /\btox\-/.test(list.className);
+
 const isChildOfBody = (editor: Editor, elm: Node): boolean => {
   return editor.dom.isChildOf(elm, editor.getBody());
 };
 
-const isTableCellNode = (node: Node | null): boolean => {
-  return Type.isNonNullable(node) && /^(TH|TD)$/.test(node.nodeName);
-};
+const matchNodeNames = <T extends Node = Node>(regex: RegExp) =>
+  (node: Node | null): node is T => Type.isNonNullable(node) && regex.test(node.nodeName);
 
-const isListNode = (editor: Editor) => (node: Node | null): boolean => {
-  return Type.isNonNullable(node) && (/^(OL|UL|DL)$/).test(node.nodeName) && isChildOfBody(editor, node);
-};
+const isListNode = matchNodeNames<HTMLOListElement | HTMLUListElement | HTMLDListElement>(/^(OL|UL|DL)$/);
+
+const isTableCellNode = matchNodeNames<HTMLTableHeaderCellElement | HTMLTableCellElement>(/^(TH|TD)$/);
+
+const inList = (editor: Editor, e: Node[], nodeName: string): boolean =>
+  Arr.findUntil(e, isListNode, isTableCellNode)
```

Comment:
This code is a bit complex, so an explanatory comment about why we stop if we find a table node or exactly why we are doing this might help the reader.

Reply:
Ah having looked at the code in `Buttons` that this was adapted from, this might be a good case for keeping the parameter name as `parents` so that it's clearer that we are traversing *up* DOM structures rather than *across*.

Reply:
Although I see that it's mostly the same code as the lists plugin, so there is value in keeping it consistent. Just things to think about in general, then. Lists called it `parents`  though, which supports that argument.

Reply:
Yeah, made a mistake, was passing the event initially. Changed it to passing the parents but didn't rename the parameter. Fixed, and thank you.